opm/parser/eclipse/EclipseState/Schedule/WellTestState.hpp

```diff
@@ -0,0 +1,82 @@
+/*
+  Copyright 2018 Statoil ASA.
+
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#ifndef WELLTEST_STATE_H
+#define WELLTEST_STATE_H
+
+#include <cstddef>
+#include <string>
+#include <vector>
+
+#include <opm/parser/eclipse/EclipseState/Schedule/WellTestConfig.hpp>
+
+namespace Opm {
+
+class WellTestState {
+public:
+    /*
+      This class implements a small mutable state object which keeps track of
+      which wells have been automatically closed by the simulator, and by
+      checking with the WTEST configuration object the class can return a list
+      (well_name,reason) pairs for wells which should be checked as candiates
+      for opening.
+    */
+
+
+
+    struct ClosedWell {
+        std::string name;
+        WellTestConfig::Reason reason;
+        double last_test;
+        int num_attempt;
+    };
+
+    /*
+      The simulator has decided to close a particular well; we then add it here
+      as a closed well with a particualar reason.
+    */
+    void addClosedWell(const std::string& well_name, WellTestConfig::Reason reason, double sim_time);
+
```

Comment:
Why provide the sim_time here? Isn't it the timing between the testing that is used.